### PR TITLE
DAOS-8586: container: snapshot commands output hex epochs

### PR DESCRIPTION
**Pull request by kccain**
In a previous change, the daos container list-snap command was changed
so its human-readable output presented epochs in hex format. However,
other commands such as create-snap continued to output base 10 values.
This was an oversight. And, the container destroy-snap command --epc
and --epcrange arguments did not properly parse hex formatted epochs.

With this change, all snapshot human-readable output is changed so that
epochs are printed in hex format with a 0x prefix. This includes
container query (ci_lsnapshot most recent snapshot). And the snapshot
commands accepting --epc or --epcrange arguments accept either decimal
or hexadecimal formatted values.

In a future change, the daos snapshot commands, when run with the -j
argument to the daos utility, will produce JSON-formatted information.
Currently the tools call a C-based handler function that produces
only human-readable output. When that change is made, daos -j callers
will then be able to take the unformatted value and pass it as the
--epc argument to other snapshot commands.

Test-tag-vm: pr daos_snapshot snap

Signed-off-by: Kenneth Cain <kenneth.c.cain@intel.com>

### CODE REVIEW

File: src/control/cmd/daos/snapshot.go [hunk 1]
```diff
@@ -18,8 +18,9 @@ import "C"
 type containerSnapshotCreateCmd struct {
 	existingContainerCmd
 
-	Epoch uint64 `long:"epc" short:"e" description:"epoch to use for snapshot"`
-	Name  string `long:"snap" short:"s" description:"snapshot name"`
+	// FIXME: epoch is not an input for snapshot create
```
Comment (mjmac) [inline src/control/cmd/daos/snapshot.go:21]: If this shouldn't be accepted by the create command, probably best to just remove it now.

**Review by mjmac — APPROVED** [review]
LGTM. Nice unit test coverage.

File: src/control/cmd/daos/flags.go [hunk 2]
```diff
@@ -33,14 +33,33 @@ func flagTestInit() (func(), error) {
 	}, nil
 }
 
+type EpochFlag struct {
+	Set bool
+	Ev  uint64
```
Comment (kjacque) [inline src/control/cmd/daos/flags.go:38]: nit - 'Value' may be a better name

File: src/control/cmd/daos/snapshot.go (hunk 1 above)
Comment (kjacque) [reply src/control/cmd/daos/snapshot.go:21]: +1

**Review by kjacque — DISMISSED** [review]
Not blocking on any of my comments, but it does seem like we could easily remove the epoch input for the create command now if it shouldn't be there.

**Comment by kccain** [comment]
Build 5 test results went well with extended testing using this commit message pragma:
Test-tag-vm: pr daos_snapshot snap offline_reintegration_daily offline_reintegration_full
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-6803/5/tests/

One failing test (from the weekly full_regression set) is container/snapshot.py that has an outstanding bug https://daosio.atlassian.net/browse/DAOS-8602 . I did some investigation of this and came up with an idea of what may be wrong in the test. @dinghwah We are co-authoring a separate PR to address that issue.

I will merge latest master, and use a different Test-tag-vm to re-run all of those tests minus the container/snapshot.py ones, and hopefully achieve another clean test run - and will get reviews concurrently to this.

File: src/control/cmd/daos/flags.go (hunk 2 above)
Comment (kccain) [reply src/control/cmd/daos/flags.go]: Changed Ev -> Value.

File: src/control/cmd/daos/snapshot.go (hunk 1 above)
Comment (kccain) [reply src/control/cmd/daos/snapshot.go]: Removed

**Review by shimizukko — APPROVED** [review]

**Pull request merged**